src/openrct2/object/RideObject.cpp

```diff
@@ -532,3 +474,642 @@ uint8 RideObject::CalculateNumHorizontalFrames(const rct_ride_entry_vehicle * ve
+            }
+            else
+            {
+                peepLoadingPositions.push_back(0);
+            }
+
+            size_t i;
+            json_t * route;
+            json_array_foreach(jLoadingWaypoints, i, route)
+            {
+                if (json_is_array(route))
+                {
+                    size_t j;
+                    json_t * waypoint;
+                    json_array_foreach(route, j, waypoint)
+                    {
+                        if (json_is_array(waypoint) && json_array_size(waypoint) >= 2)
+                        {
+                            auto x = json_integer_value(json_array_get(waypoint, 0));
+                            auto y = json_integer_value(json_array_get(waypoint, 1));
+                            peepLoadingPositions.push_back(x);
+                            peepLoadingPositions.push_back(y);
+                        }
+                    }
+                    peepLoadingPositions.push_back(0);
+                    peepLoadingPositions.push_back(0);
+                }
+            }
+        }
+    }
+
+    auto jFrames = json_object_get(jCar, "frames");
+    car.sprite_flags = ObjectJsonHelpers::GetFlags<uint16>(jFrames, {
+        { "flat", VEHICLE_SPRITE_FLAG_FLAT },
+        { "gentleSlopes", VEHICLE_SPRITE_FLAG_GENTLE_SLOPES },
+        { "steepSlopes", VEHICLE_SPRITE_FLAG_STEEP_SLOPES },
+        { "verticalSlopes", VEHICLE_SPRITE_FLAG_VERTICAL_SLOPES },
+        { "diagonalSlopes", VEHICLE_SPRITE_FLAG_DIAGONAL_SLOPES },
+        { "flatBanked", VEHICLE_SPRITE_FLAG_FLAT_BANKED },
+        { "inlineTwists", VEHICLE_SPRITE_FLAG_INLINE_TWISTS },
+        { "flatToGentleSlopeBankedTransitions", VEHICLE_SPRITE_FLAG_FLAT_TO_GENTLE_SLOPE_BANKED_TRANSITIONS },
+        { "diagonalGentleSlopeBankedTransitions", VEHICLE_SPRITE_FLAG_DIAGONAL_GENTLE_SLOPE_BANKED_TRANSITIONS },
+        { "gentleSlopeBankedTransitions", VEHICLE_SPRITE_FLAG_GENTLE_SLOPE_BANKED_TRANSITIONS },
+        { "gentleSlopeBankedTurns", VEHICLE_SPRITE_FLAG_GENTLE_SLOPE_BANKED_TURNS },
+        { "flatToGentleSlopeWhileBankedTransitions", VEHICLE_SPRITE_FLAG_FLAT_TO_GENTLE_SLOPE_WHILE_BANKED_TRANSITIONS },
+        { "corkscrews", VEHICLE_SPRITE_FLAG_CORKSCREWS },
+        { "restraintAnimation", VEHICLE_SPRITE_FLAG_RESTRAINT_ANIMATION },
+        { "curvedLiftHill", VEHICLE_SPRITE_FLAG_CURVED_LIFT_HILL },
+        { "VEHICLE_SPRITE_FLAG_15", VEHICLE_SPRITE_FLAG_15 } });
+
+    car.flags |= ObjectJsonHelpers::GetFlags<uint32>(jCar, {
+        { "VEHICLE_ENTRY_FLAG_POWERED_RIDE_UNRESTRICTED_GRAVITY", VEHICLE_ENTRY_FLAG_POWERED_RIDE_UNRESTRICTED_GRAVITY },
+        { "VEHICLE_ENTRY_FLAG_NO_UPSTOP_WHEELS", VEHICLE_ENTRY_FLAG_NO_UPSTOP_WHEELS },
+        { "VEHICLE_ENTRY_FLAG_NO_UPSTOP_BOBSLEIGH", VEHICLE_ENTRY_FLAG_NO_UPSTOP_BOBSLEIGH },
+        { "VEHICLE_ENTRY_FLAG_MINI_GOLF", VEHICLE_ENTRY_FLAG_MINI_GOLF },
+        { "VEHICLE_ENTRY_FLAG_4", VEHICLE_ENTRY_FLAG_4 },
```

Comment:
What happens if ever find the purpose of this flag and rename it? All the objects would have to be updated?

Reply:
Yes all objects would be updated after a proper name is found.